New git file finder that support symlinks.

sbidoul opened the pull request:
Closes #245

sbidoul committed:
New git file finder that support symlinks.

RonnyPfannschmidt commented:
it seems this breaks the expectations of some tests which is ok,
part of it is pip 10.x breaking the world for bad code (and i had some of that, sorry)

sbidoul committed:
Fix pip invocation test.

sbidoul committed:
Adapt git_file_finder tests style

sbidoul committed:
Fix existing git file finder test

Make the test compliant with file finder spec
(although integration.find_files is not spec compliant yet,
because it defaults to '.' with path is empty)

sbidoul commented:
No need to be sorry about anything. The work behing done here is far from easy, with huge expectations from the world.

It's green now.

RonnyPfannschmidt commented on setuptools_scm/git_file_finder.py:
```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2018 ACSONE SA/NV
+# License LGPLv3 (http://www.gnu.org/licenses/lgpl-3.0-standalone.html)
```
i just noted the licence, is it acceptable for you to go to MIT?

sbidoul replied:
yep, sorry, copy/paste error, I remove that

RonnyPfannschmidt reviewed (commented):
generally well done, this is how i like a change :grinning: :+1: 

please take a look at the licensing detail

sbidoul committed:
Remove file headers to follow local style

RonnyPfannschmidt commented:
wow, that's quick ^^

sbidoul commented:
Thanks.

Is appveyor running tests on windows? I don't find the link.

RonnyPfannschmidt commented:
i need to check to see why its not there

RonnyPfannschmidt commented:
it seems a configuration issue is preventing the ci run, it'll take a bit to investigate

RonnyPfannschmidt commented:
indeed, it got strangely messed up in a token cleanup and appveyor never told details until i recreated the project

sbidoul committed:
Skip symlink tests on Windows

sbidoul committed:
Add file name case test

sbidoul committed:
Normalize case when looking up file in git file list

sbidoul committed:
Handle windows path separators in tests

RonnyPfannschmidt commented:
it seems like windows is giving and giving :disappointed:

sbidoul commented:
@RonnyPfannschmidt a bit of trial and error indeed :)

I think I nailed it now. Basically it boils down to using normcase everywhere when looking up in the git files list. 

I chose to preserve case and separators in the finder output.

The last two failing tests also fail on master.

RonnyPfannschmidt commented:
thanks for finalizing the details, i'll take a look at the rest

RonnyPfannschmidt commented:
ok, as far as i understand this is merge-able and the tests that dont work in windows are my fault due to developing while windows ci was accidentally off

please ack then i'll merge

sbidoul commented:
yes, this PR is ready to merge

RonnyPfannschmidt commented:
great job, thanks for jumping all the hops to iron that one out
